modules/gateway/outputs.tf

```diff
@@ -24,10 +24,14 @@ output "public_ip_address_fqdn" {
 
 output "gateway_id" {
   description = "Identifier of the application gateway for GraphDB"
-  value       = var.gateway_enable_private_access ? azurerm_application_gateway.graphdb-private[0].id : azurerm_application_gateway.graphdb-public[0].id
+  value       = var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-private) > 0 ? azurerm_application_gateway.graphdb-private[0].id : !var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-public) > 0 ? azurerm_application_gateway.graphdb-public[0].id : null
 }
 
 output "gateway_backend_address_pool_id" {
   description = "Identifier of the application gateway backend address pool"
-  value       = var.gateway_enable_private_access ? one(azurerm_application_gateway.graphdb-private[0].backend_address_pool).id : one(azurerm_application_gateway.graphdb-public[0].backend_address_pool).id
+  value = (var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-private) > 0 && length(azurerm_application_gateway.graphdb-private[0].backend_address_pool) > 0
```

Comment:
That's a mouthful at least leave a comment what it does

Reply:
Added comment and a change to this output.